sale_credit_point/models/partner.py

```diff
@@ -63,3 +69,24 @@ def credit_point_increase(self, amount, comment=''):
 
     def credit_point_decrease(self, amount, comment=''):
         self._credit_point_update(self.credit_point - amount, comment=comment)
+
+    def update_history(self, amount, operation, comment):
+        history_model = self.env["credit.point.history"]
+        vals = {
+            'partner_id': self.id,
+            'operation': operation,
+            'amount': amount,
+            'comment': comment,
+        }
+        return history_model.create(vals)
+
+    def _compute_yearly_point_increase(self):
+        history = self.env["credit.point.history"]
+        current_year = date.today().year
+        for record in self:
+            # TODO: is search_read better in such case?
```

Comment:
yes, read only what you need. OR, as you are already bypassing ORM security, you can even use raw SQL query (you can even group by partner_id and do only one select too, if you prefer)